addons/core/functions/fnc_requestRadios.sqf

```diff
@@ -27,68 +31,118 @@ private _lastExec = GVAR(VehicleConfigCacheNamespace) getVariable "TFAR_fnc_requ
+                    _OldItem = (call TFAR_fnc_getDefaultRadioClasses) param [[2, 1] select ((TFAR_givePersonalRadioToRegularSoldier) or {leader _requestedUnit == _requestedUnit} or {rankId _requestedUnit >= 2}), ""];
+                };
+                private _NewItem = format["%1_%2", [_OldItem, "tf_parent", ""] call DFUNC(getWeaponConfigProperty), _newID];
+                _requestedUnit linkItem _NewItem;
+
+                _newRadios pushBack _NewItem;
+
+                private _settings = _TF_SettingsToCopy deleteAt _index;
+                if ([_NewItem, _OldItem] call TFAR_fnc_isSameRadio) then {
+                    private _localSettings = TFAR_RadioSettingsNamespace getVariable (format["%1_local", _settings]);
+                    if !(isNil "_localSettings") then {
+                        [_NewItem, _OldItem, true] call TFAR_fnc_setSwSettings;
                     };
                 };
-                (_copyIndex + 1)
+
+            } else {
+                _requestedUnit unassignItem _OldItem;
+                _requestedUnit removeItem _OldItem;
+            };
+        };
+
+
+        private _OldItem = _radiosToReplace deleteAt 0;
+        {
+            private _OldItem = _x;
+            private _newID = _response deleteAt 0;
+
+            _requestedUnit removeItem _OldItem;
+
+            If (_OldItem == "ItemRadio") then {
+                _OldItem = (call TFAR_fnc_getDefaultRadioClasses) param [[2, 1] select ((TFAR_givePersonalRadioToRegularSoldier) or {leader _requestedUnit == _requestedUnit} or {rankId _requestedUnit >= 2}), ""];
             };
-            private _linkFirstItem = TFAR_RadioReqLinkFirstItem;
-            if (_linkFirstItem) then {
-                TFAR_RadioReqLinkFirstItem = false;
-                TFAR_currentUnit linkItem (_response select 0);
-                _copyIndex = [_settingsCount, _copyIndex, (_response select 0),_TF_SettingsToCopy] call _fnc_CopySettings;
-                [(_response select 0), getPlayerUID player, true] call TFAR_fnc_setRadioOwner;
+
+            private _NewItem = format["%1_%2", [_OldItem, "tf_parent", ""] call DFUNC(getWeaponConfigProperty), _newID];
+
+            If (_requestedUnit canAdd _NewItem) then {
+                _requestedUnit AddItem _NewItem;
+                _newRadios pushBack _NewItem;
             };
 
-            {
-                TFAR_currentUnit addItem _x;
-                _copyIndex = [_settingsCount, _copyIndex, _x,_TF_SettingsToCopy] call _fnc_CopySettings;
-                [_x, getPlayerUID player, true] call TFAR_fnc_setRadioOwner;
-            } forEach (if (_linkFirstItem) then {_response select [1]} else {_response});
-        };
-    } else {
-        hintC _response;
-    };
-    call TFAR_fnc_hideHint;
-    //								unit, radios
-    ["OnRadiosReceived", [TFAR_currentUnit, _response]] call TFAR_fnc_fireEventHandlers;
+            private _settings = _TF_SettingsToCopy deleteAt _index;
```

Comment:
Rename that var to `_settingsToCopy`

Reply:
That variable is undefined. https://github.com/michail-nikolaev/task-force-arma-3-radio/pull/1362/files/e27fa0247429a0fb0713cbd30ecb5753ce505028#diff-df71d93b827ddd11118272b9ded5cb17R34
https://github.com/michail-nikolaev/task-force-arma-3-radio/pull/1362/files/e27fa0247429a0fb0713cbd30ecb5753ce505028#diff-cc31dbeae3ba49f9dcc5b4c744b3cbc6R18
seems like that was changed a long time ago.

Reply:
https://github.com/Dorbedo/task-force-arma-3-radio/blame/362742dde138d2c3688e9e4b43f18fe696825b88/addons/core/functions/fnc_radioToRequestCount.sqf Old version

@Dorbedo

Reply:
why didn't you change it with your commit?

Reply:
because I didn't know what to change it to.
Do we want to copy all settings of all previously initialized radios?
should the func that returns the list of radios to instantiate additionally return the list of _TF_SettingsToCopy like it did previously?